### PR TITLE
fix linux build to include gcc flag -Wall

### PR DESCRIPTION
**Pull request by jebrando**

### CODE REVIEW

File: tests/mu_expand_no_expand_test.c [hunk 1]
```diff
@@ -9,6 +9,11 @@
 
 #include "mu_expand_no_expand_test.h"
 
+// For gcc build that causes a false negative
+#ifndef _MSC_VER
+#pragma GCC diagnostic ignored "-Wunused-value"
```
Comment (mattdurak) [inline tests/mu_expand_no_expand_test.c:14]: did you fix this with the "(void)variable_name" in the other file? Just wondering if it's an easy fix to be explicit vs. disabling the warning. /consider

File: build/linux/build_linux.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+build_folder=$build_root"/cmake_linux"
+
+# Set the default cores
+CORES=$(grep -c ^processor /proc/cpuinfo 2>/dev/null || sysctl -n hw.ncpu)
+
+rm -r -f $build_folder
+mkdir -p $build_folder
+pushd $build_folder
+cmake -Drun_valgrind:BOOL=ON -Drun_helgrind:BOOL=ON -Drun_drd:BOOL=ON -Drun_int_tests:bool=ON $build_root
```
Comment (mattdurak) [inline build/linux/build_linux.sh:18]: >cmake [](http://example.com/codeflow?start=0&length=5)

missing yml update to actually call this in a pipeline?

**Review by mattdurak — APPROVED** [review]
:shipit:

File: tests/mu_expand_no_expand_test.c (hunk 1 above)
Comment (jebrando) [reply tests/mu_expand_no_expand_test.c:14]: The unused variable in this file is in a macro that gets constructed so I couldn't just add a void to it.

**Pull request merged**